packages/cw721/src/state.rs

```diff
@@ -0,0 +1,241 @@
+use serde::Serialize;
+
+/// Creator owns this contract and can update collection info!
+/// !!! Important note here: !!!
+/// - creator is stored using using cw-ownable's OWNERSHIP singleton, so it is not stored here
+/// - in release v0.18.0 it was used for minter (which is confusing), but now it is used for creator
+pub const CREATOR: OwnershipStore = OwnershipStore::new(OWNERSHIP_KEY);
+/// - minter is stored in the contract storage using cw_ownable::OwnershipStore (same as for OWNERSHIP but with different key)
+pub const MINTER: OwnershipStore = OwnershipStore::new("collection_minter");
+
+/// Default CollectionInfoExtension with RoyaltyInfo
+pub type DefaultOptionCollectionInfoExtension = Option<CollectionInfoExtension<RoyaltyInfo>>;
+pub type DefaultOptionMetadataExtension = Option<Metadata>;
+
+pub struct Cw721Config<
+    'a,
+    TMetadata,
+    TCustomResponseMessage,
+    TExtensionExecuteMsg,
+    TMetadataResponse,
+    TCollectionInfoExtension,
+> where
+    TMetadata: Serialize + DeserializeOwned + Clone,
+    TMetadataResponse: CustomMsg,
+    TExtensionExecuteMsg: CustomMsg,
+    TCollectionInfoExtension: Serialize + DeserializeOwned + Clone,
+{
+    /// Note: do not use deprecated/legacy key "nft_info"!
+    pub collection_info: Item<'a, CollectionInfo<TCollectionInfoExtension>>,
+    pub token_count: Item<'a, u64>,
+    /// Stored as (granter, operator) giving operator full control over granter's account.
+    /// NOTE: granter is the owner, so operator has only control for NFTs owned by granter!
+    pub operators: Map<'a, (&'a Addr, &'a Addr), Expiration>,
+    /// Note: do not use deprecated/legacy keys "tokens" and "tokens__owner"!
+    pub nft_info: IndexedMap<'a, &'a str, NftInfo<TMetadata>, TokenIndexes<'a, TMetadata>>,
+    pub withdraw_address: Item<'a, String>,
+
+    pub(crate) _custom_response: PhantomData<TCustomResponseMessage>,
+    pub(crate) _custom_query: PhantomData<TMetadataResponse>,
+    pub(crate) _custom_execute: PhantomData<TExtensionExecuteMsg>,
+}
+
+impl<
+        TMetadata,
+        TCustomResponseMessage,
+        TExtensionExecuteMsg,
+        TMetadataResponse,
+        TCollectionInfoExtension,
+    > Default
+    for Cw721Config<
+        'static,
+        TMetadata,
+        TCustomResponseMessage,
+        TExtensionExecuteMsg,
+        TMetadataResponse,
+        TCollectionInfoExtension,
+    >
+where
+    TMetadata: Serialize + DeserializeOwned + Clone,
+    TExtensionExecuteMsg: CustomMsg,
+    TMetadataResponse: CustomMsg,
+    TCollectionInfoExtension: Serialize + DeserializeOwned + Clone,
+{
+    fn default() -> Self {
+        Self::new(
+            "collection_info", // Note: do not use deprecated/legacy key "nft_info"
+            "num_tokens",
+            "operators",
+            "nft",        // Note: do not use deprecated/legacy key "tokens"
+            "nft__owner", // Note: do not use deprecated/legacy key "tokens__owner"
+            "withdraw_address",
+        )
+    }
```

Comment:
Pay ATTENTION to these key renamings in storage:
nft_info -> collection_info
tokens -> nft
tokens__owner -> nft__owner

This is covered in migration is as well. Main reason for renaming is that collection info (formerly: contract info) is a new struct with collection info extension!

Instead of using and overriding(!) existing "nft_info" store, a new store `collection_info` is used. This way if something goes wrong it'll be possible reverting back to previous version where old stores still exist!

Reply:
This has been updated, only "collection_info" key is new, reverted back:
"nft" -> "tokens"
"nft__owner" -> "tokens__owner"
Because it will impact migration and gas fees without any benefit.